### ISSUE TITLE
NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE false positive?

### ISSUE THREAD

**Issue by xenoterracide** [spotbugs/spotbugs#1530]
spotbugs is returning `NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE` at the very last line of this method. I don't see any way a null pointer there is possible. Though maybe I'm missing something.

```java
  @SuppressWarnings("checkstyle:ReturnCount")
  Optional<Path> findConfig( String extension, Path searchDir ) {
    var configFile = searchDir.resolve( ".config" )
      .resolve( "brix" )
      .resolve( this.pathFromConfigDir( extension ) );

    log.trace( "searching for: '{}'", configFile );
    if ( Files.exists( configFile ) ) {
      log.debug( "found: '{}'", configFile );
      return Optional.of( configFile );
    }

    if ( Objects.equals( home, searchDir ) ) {
      return Optional.empty();
    }
    return this.findConfig( extension, searchDir.getParent() );
  }
```

```
Fedora 33
5.11.10-200.fc33.x86_64
 13:31:59 up 24 days, 14:37,  1 user,  load average: 0.51, 0.86, 0.94
              total        used        free      shared  buff/cache   available
Mem:            15G         11G        1.7G        998M        2.6G        3.0G
Swap:           12G        3.2G        8.9G
```
```
------------------------------------------------------------
Gradle 7.0
------------------------------------------------------------

Build time:   2021-04-09 22:27:31 UTC
Revision:     d5661e3f0e07a8caff705f1badf79fb5df8022c4

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.10 (AdoptOpenJDK 11.0.10+9)
OS:           Linux 5.11.10-200.fc33.x86_64 amd64
```

```
+--- com.github.spotbugs:spotbugs:{strictly 4.2.1} -> 4.2.1 (c)
```

**Comment by ThrawnCA** [comment]
Well, a null pointer seems possible on the first line, and the last line recursively calls it, so...

**Comment by xenoterracide** [comment]
it's not though, but maybe I'm thinking of how checker and error prone nullaway work, they infer non null by default unless annotated with `@Nullable`

**Comment by xenoterracide** [comment]
yeah closing this in favor of https://github.com/typetools/checker-framework/issues/4584 sorry
